### ISSUE TITLE
[Feature]: Compatibility with sqlite-zstd

### ISSUE THREAD

**Issue by yfdyh000** [sqlitebrowser/sqlitebrowser#3244]
### Describe the new feature

https://github.com/phiresky/sqlite-zstd

The extension can be loaded in the software, but some features seem to be incompatible.

1. `SELECT
zstd_enable_transparent('{"table": "objects", "column": "data1", "compression_level": 19, "dict_chooser": "''a''"}');` will be an error, due to the SQL execution will bring a transaction (`SAVEPOINT "RESTOREPOINT";`).
```
Could not start transaction

Caused by:
    0: cannot start a transaction within a transaction
    1: Error code 1: SQL error or missing database
```

2. `while table_name will be a view that can be queried as normally, including SELECT, INSERT, UPDATE, and DELETE queries.` from sqlite-zstd, but the software refuses to modify a view via UI controls (disabled).

**Comment by chrisjlocke** [comment]
There is no guarantee DB4S works with all extensions - it's just impossible to work nicely with everything.

The 'could not start a transaction' is because we start a restore point before issuing statements.  It looks like the zstd does the same thing, and you can't nest transactions.
Unfortunately, this is not something that can be turned off within DB4S - it's just how it works.

Views can be edited in DB4S, but I'm not sure it'll work.  

![image](https://user-images.githubusercontent.com/7872218/206840768-9900e964-61b6-4ce5-afc5-35dbbf98362c.png)

If you right-click the table header when viewing a view, you can select the menu option, 'unlock view editing'.
There is still a bit of work to be done to get this working - details are in the wiki - but see if that helps.

**Comment by mgrojo** [comment]
We actually start a [SAVEPOINT](https://www.sqlite.org/lang_savepoint.html), not a transaction.  Since savepoints can be named and nested, I suppose that if they also used that we could work together. @yfdyh000, do you know if there are technical reasons for [sqlite-zstd](https://github.com/phiresky/sqlite-zstd) having chosen a full transaction instead of a savepoint?

**Comment by yfdyh000** [comment]
> We actually start a [SAVEPOINT](https://www.sqlite.org/lang_savepoint.html), not a transaction. Since savepoints can be named and nested, I suppose that if they also used that we could work together. @yfdyh000, do you know if there are technical reasons for [sqlite-zstd](https://github.com/phiresky/sqlite-zstd) having chosen a full transaction instead of a savepoint?

No, I am a novice for the library and did not see a technical reason.
@phiresky

**Comment by phiresky** [comment]
I was not aware of savepoints so there's no particular why they couldn't be used in sqlite-zstd. I guess it should just be using [Connection::savepoint](https://docs.rs/rusqlite/latest/rusqlite/struct.Connection.html#method.savepoint) instead of transaction in this line: https://github.com/phiresky/sqlite-zstd/blob/6d8b8b1b2f225b13636128bbe03a69e28adccb6a/src/transparent.rs#L167

Assuming savepoint doesn't change the semantics. 

I've used sqlite-zstd with sqlitebrowser and most stuff worked, but I hadn't tried the zstd_enable_transparent function I guess.

Note that sqlite-zstd is not affiliated with the zstd project (just for clarification)

**Comment by mgrojo** [comment]
> I was not aware of savepoints so there's no particular why they couldn't be used in sqlite-zstd. I guess it should just be using [Connection::savepoint](https://docs.rs/rusqlite/latest/rusqlite/struct.Connection.html#method.savepoint) instead of transaction in this line: https://github.com/phiresky/sqlite-zstd/blob/6d8b8b1b2f225b13636128bbe03a69e28adccb6a/src/transparent.rs#L167
> 

You'd have to change also the commit to a release, but I haven't seen it on the Rust library. Otherwise, using `zstd_enable_transparent` will also save the changes the user has previously done in the database to the disk, which will be inconsistent with the general sqlitebrowser behaviour. 

> Assuming savepoint doesn't change the semantics.

As far as I know, the only difference between savepoints and regular transactions is that the former can be nested. Note that if you release a non-nested savepoint, the changes are immediately committed, that is, it behaves as a `BEGIN TRANSACTION; ... ; COMMIT`.

**Comment by phiresky** [comment]
Looks like the Savepoint::commit() method in rusqlite actually calls release with the correct (generated) name so no change needed :)

I've opened an issue here https://github.com/phiresky/sqlite-zstd/issues/24 but I'm not sure if I'll get around to it myself.
